src/Compilers/VisualBasic/Portable/BoundTree/Expression.vb

```diff
@@ -1382,6 +1370,105 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         End Function
     End Class
 
+    Partial Class BoundFieldOrPropertyInitializer
+        Implements IFieldInitializer, IPropertyInitializer
```

Comment:
BoundFieldOrPropertyInitializer gets generated in most but apparently not all circumstances. This node is generated, I think, for normal compilation as well as for semantic models, so I am reluctant to try to figure out how to elide them.

If used in the expected pattern--check Kind and then cast based on Kind--things will work out OK. If used by testing for type rather than testing for Kind, then confusion is likely.

Given that this node type is used by the compiler, it's probably simpler to use it for all = Value constructs rather than introducing Bound(Property/Field)EqualsValue, if there is a desire to avoid partially-overlapping node types.